From 2d0b7dbb82e176ae632ae36998152aebb0fc2d56 Mon Sep 17 00:00:00 2001
From: TheWhisp <daavis.90@gmail.com>
Date: Sat, 2 Nov 2013 18:09:06 +0200
Subject: [PATCH] add option to build LPA support in addition to legacy OMX

LPA works on msm7x27a devices which uses ION.
However, legacy OMX stuff is still needed on these targets.

Change-Id: I3e104c005a6522ce314a96045930935aa322ea7f
---
 media/libmedia/Android.mk        |  4 ++++
 media/libstagefright/Android.mk  | 10 +++++++++-
 services/audioflinger/Android.mk |  4 ++++
 3 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/media/libmedia/Android.mk b/media/libmedia/Android.mk
index b662ffe..ce67af0 100644
--- a/media/libmedia/Android.mk
+++ b/media/libmedia/Android.mk
@@ -92,6 +92,10 @@ endif
 endif
 endif
 
+ifeq ($(TARGET_QCOM_LEGACY_OMX_WITH_LPA),true)
+   LOCAL_CFLAGS += -DQCOM_ENHANCED_AUDIO
+endif
+
 # for <cutils/atomic-inline.h>
 LOCAL_CFLAGS += -DANDROID_SMP=$(if $(findstring true,$(TARGET_CPU_SMP)),1,0)
 LOCAL_SRC_FILES += SingleStateQueue.cpp
diff --git a/media/libstagefright/Android.mk b/media/libstagefright/Android.mk
index 628bed3..a0e3bf0 100644
--- a/media/libstagefright/Android.mk
+++ b/media/libstagefright/Android.mk
@@ -114,6 +114,15 @@ ifneq ($(filter caf bfam,$(TARGET_QCOM_AUDIO_VARIANT)),)
     LOCAL_CFLAGS += -DQCOM_ENHANCED_AUDIO
 endif
 
+ifeq ($(TARGET_QCOM_LEGACY_OMX_WITH_LPA),true)
+    LOCAL_SRC_FILES  += QCMediaDefs.cpp
+    LOCAL_SRC_FILES += LPAPlayer.cpp
+    LOCAL_CFLAGS += -DLEGACY_LPA
+    LOCAL_CFLAGS += -DQCOM_ENHANCED_AUDIO
+    LOCAL_CFLAGS += -DQCOM_LEGACY_MMPARSER
+    LOCAL_CFLAGS += -DQCOM_LEGACY_OMX
+endif
+
 ifneq ($(TARGET_QCOM_MEDIA_VARIANT),)
 LOCAL_C_INCLUDES += \
         $(TOP)/hardware/qcom/media-$(TARGET_QCOM_MEDIA_VARIANT)/mm-core/inc
@@ -210,7 +219,6 @@ ifeq ($(TARGET_QCOM_LEGACY_OMX),true)
     LOCAL_CFLAGS += -DQCOM_LEGACY_MMPARSER
     LOCAL_SRC_FILES  += QCMediaDefs.cpp
 endif
-
 include $(BUILD_SHARED_LIBRARY)
 
 include $(call all-makefiles-under,$(LOCAL_PATH))
diff --git a/services/audioflinger/Android.mk b/services/audioflinger/Android.mk
index c0d56d7..3b16c74 100644
--- a/services/audioflinger/Android.mk
+++ b/services/audioflinger/Android.mk
@@ -17,6 +17,10 @@ ifneq ($(filter caf bfam,$(TARGET_QCOM_AUDIO_VARIANT)),)
     LOCAL_CFLAGS += -DQCOM_ENHANCED_AUDIO
 endif
 
+ifeq ($(TARGET_QCOM_LEGACY_OMX_WITH_LPA),true)
+   LOCAL_CFLAGS += -DQCOM_ENHANCED_AUDIO
+endif
+
 LOCAL_SRC_FILES:=               \
     AudioFlinger.cpp            \
     Threads.cpp                 \
-- 
1.8.1.2

